docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -1,20 +1,125 @@
 
 
+Mechanism
+--------------
+The NeMo Distributed Checkpoint enables saving and loading models from multiple ranks in parallel. It employs a novel strategy **Fully Parallel Saving** (FPS) to partition the optimizer states, gradients and model parameters across all GPU ranks. When saving the checkpoint of a distributed optimizer, each DP rank holds its shard of the optimizer state and independently writes its shard to the shared storage (grad buffer).
+
+When loading the checkpoint, each DP rank reads its corresponding checkpoint file (shard) to recover. If different parallelism strategies are needed (e.g., tensor parallelism, pipeline parallelism), each rank can also access other checkpoint files to transfer data to the correct locations. 
+
+NeMo enables users to resume training from a checkpoint saved with different tensor and pipeline parallelism degrees, providing the flexibility to change training configurations as needed during training. 
+
+The following figure illustrates fully parallel saving in NeMo Framework, utilizing data-parallel replicas for writing across nodes.
+
+.. image:: imgs/dist_ckpt_0.png
+
+
+*Figure 1. Fully parallel saving in NeMo Framework uses the data-parallel replicas for parallel writing across nodes*
+
+The following figure illustrates asynchronous saving in NeMo Framework, where checkpoints are saved in the background while training continues. Asynchronous parallel saving allows model parameters to be copied to the CPU first before persisting the checkpoint to stable storage in the background. This process minimizes interruptions to the main training, thereby speeding up the distributed checkpointing process.
+
+.. image:: imgs/dist_ckpt_1.png
+
+
+*Figure 2. Asynchronous saving in Megatron Core saves checkpoint at the background in parallel with training*
+
+
+Parameter Tuning
+--------------
+
+Distributed checkpoints in NeMo could be configured for pre-training and fine-tuning jobs. 
+
+In the `NeMo 1.0 YAML config file <https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/migration/checkpointing.html>`__ or `NeMo 2.0 MegatronStrategy <https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/migration/checkpointing.html>`__, you can enable and tune these parameters.
+
+The latest NeMo version is Nemo 2.0 (NGC container ``nvcr.io/nvidia/nemo:24.09``).
+
+
+Best Practices
+^^^^^^^^^^^
+
+Here are best practices for configuring distributed checkpoints in NeMo:
+
+.. code-block:: python
+
+        	dist_ckpt_format: 'torch_dist'
+
+        	dist_ckpt_load_on_device: True
+
+        	dist_ckpt_parallel_save: True
+
+        	dist_ckpt_parallel_save_within_dp: False
+
+        	dist_ckpt_parallel_load: True
+
+        	dist_ckpt_torch_dist_multiproc: 2
+
+        	dist_ckpt_assume_constant_structure: False
+
+        	dist_ckpt_parallel_dist_opt: True
+
+        	dist_ckpt_load_strictness: null
+
+
+Here are more details of the checkpoint format options and related parameters:
```

Comment:
revise

Here's a summary of the checkpoint format options and related parameters: